src/contract_spec.ts

```diff
@@ -687,11 +765,323 @@ function findCase(name: string) {
   };
 }
 
-function readObj(args: object, input: xdr.ScSpecFunctionInputV0): any {
-  let inputName = input.name().toString();
-  let entry = Object.entries(args).find(([name, _]) => name === inputName);
-  if (!entry) {
-    throw new Error(`Missing field ${inputName}`);
+const PRIMITIVE_DEFINITONS = {
+  U64: {
+    type: 'string',
+    pattern: '^[0-9]+$',
+    minLength: 1,
+    maxLength: 20 // 64-bit max value has 20 digits
+  },
+  I64: {
+    type: 'string',
+    pattern: '^-?^[0-9]+$',
+    minLength: 1,
+    maxLength: 21 // Includes additional digit for the potential '-'
+  },
+  U32: {
+    type: 'integer',
+    minimum: 0,
+    maximum: 4294967295
+  },
+  I32: {
+    type: 'integer',
+    minimum: -2147483648,
+    maximum: 2147483647
+  },
+  U128: {
+    type: 'string',
+    pattern: '^[0-9]+$',
+    minLength: 1,
+    maxLength: 39 // 128-bit max value has 39 digits
+  },
+  I128: {
+    type: 'string',
+    pattern: '^-?[0-9]+$',
+    minLength: 1,
+    maxLength: 40 // Includes additional digit for the potential '-'
+  },
+  U256: {
+    type: 'string',
+    pattern: '^[0-9]+$',
+    minLength: 1,
+    maxLength: 78 // 256-bit max value has 78 digits
+  },
+  I256: {
+    type: 'string',
+    pattern: '^-?[0-9]+$',
+    minLength: 1,
+    maxLength: 79 // Includes additional digit for the potential '-'
+  },
+  Address: {
+    type: 'string',
+    description: 'Address can be a public key or contract id'
```

Comment:
does this deserve a regex, as well?

```suggestion
    description: 'Address can be a public key or contract id',
    pattern: '^(G|C)[A-Z0-9]{55}$'
```